aur2ccr.8
'\" t
.\"     Title: aur2ccr
.\"    Author: [see the "Authors" section]
.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
.\"      Date: 04/03/2012
.\"    Manual: aur2ccr Manual
.\"    Source: aur2ccr
.\"  Language: English
.\"
.TH "aur2ccr" "8" "04/03/2012" "aur2ccr" "aur2ccr Manual"
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
aur2ccr \- simple script to automate pulling packages from AUR and Arch repos, and converting them to a format ready for uploading to CCR.
.SH "SYNOPSIS"
.sp
\fIaur2ccr\fR [options] [packages]
.SH "DESCRIPTION"
.sp
aur2ccr is a bash script to make converting packages from Arch User Repositories (AUR) or the main Arch Linux repositories to Chakra Community Repositories (CCR) easier\&. It allows you to turn any AUR *.src.tar.gz package into a CCR-ready *.src.tar.gz package, using makepkg(8), ccr(8), pacman(8), and other tools\&. 
.SH "OPTIONS"
.PP
\fB\-m\fR <\'My Name (nick) <myemail@example.com>\'>
.RS 4
Tells the script to automatically add specified maintainer information to the packages\&. Long option: --maintainer\&.
.RE

.PP
\fB\-e\fR
.RS 4
Edit each PKGBUILD with $EDITOR before building the package\&. Long option: --edit\&.
.RE

.PP
\fB\-c\fR
.RS 4
Consolidate all *.src.tar.gz packages built by the script into the working directory\&. Long option: --consolidate\&.
.RE

.PP
\fB\-i\fR
.RS 4
Install all packages build by the script, in order to test them beofre uploading to CCR\&. Long option: --install\&.
.RE

.PP
\fB\-1\fR
.RS 4
Reset the 'pkgrel' variable to 1\&. Long option: --reset-pkgrel\&.
.RE

.PP
\fB\-f\fR <\'AUR/CCR-compat source package filename\'>
.RS 4
Specify a local package to build from\&. Long option: --from-src\&.
.RE

.PP
\fB\-u\fR <\'URL of AUR/CCR-compatible *.src.tar.gz package\'>
.RS 4
Download and build a packge from a user-specified URL, instead of searching for a pacmage name in AUR and Arch repos\&. Long option: --from-url\&.
.RE

.PP
\fB\-d\fR <\'directory containing extracted *.src.tar.gz package\'>
.RS 4
Build a package using a PKGBUILD and other (optional) files in a user-specified directory, instead of searching for a pacmage name in AUR and Arch repos\&. Long option: --from-dir\&.
.RE

.PP
\fB\-C\fR <\'directory to use as working dir instead of current dir\'>
.RS 4
Work in in a directory other than the current one\&. This is similar to the -C <dir> option to tar\&. If -c or -t are set, this will put final packages in the folder given to -C\&.  Long option: --cd\&.
.RE

.PP
\fB\-t\fR
.RS 4
Build packages in $tmpdir (/tmp/aur2ccr by default), and move finished packages to the working dir, like -c\&. Don't use both -c and -t at the same time\&. Long option: --temp\&.
.RE

.PP
\fB\-l\fR <\'logfile'\>
.RS 4
Log the output of aur2ccr ONLY (not wget, makepkg, pacman, or others) to <logfile>, using tee(1)\&. Long option: --log\&.
.RE

.PP
\fB\-s\fR 
.RS 4
Perform some setup-related tasks, such as choosing the closest pacman mirrors\&. Long option: --setup\&.
.RE

.PP
\fB\-\-search\fR 
.RS 4
Search through the Arch User Repository (AUR)\&. (No short option\&.)
.RE

.PP
\fB\-a\fR 
.RS 4
Add an aur2ccr 'advertisement', in the form: # Contributor: aur2ccr (http://ddg.gg/?q=!ccr+aur2ccr)\&. Long option: --addad\&.
.RE

.PP
.SH "EXAMPLE USAGE"
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
Download and build the AUR or Arch-repo package 'pkgname', changing the maintainer information: 
aur2ccr \-m "My Name (nick) <myemail@example.com>" pkgname
.RE

.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
Download and build the AUR package 'aur-package', editing the PKGBUILD with $EDITOR before building (if $EDITOR is unset, aur2ccr will use vi): 
aur2ccr \-e aur-package
.RE

.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
Download and build 'pkgname1', 'aurpkg2', and 'archpkg3' in /tmp, moving all build source packages to the working directory: 
aur2ccr \-t pkgname1 aurpkg2 archpkg3
.RE

.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
Download and build the AUR/CCR-compatible source package (.src.tar.gz) at 'http://someurl/sourcepackage':
aur2ccr \-u 'http://someurl/sourcepackage'
.RE

.sp
.SH "CONFIGURATION"
.sp
Arch to Chakra package name translations are in /etc/aur2ccr/names.conf. Entries are in the format: 
archpkgname=chakrapkgname
.sp
Configuration options for aur2ccr are in /etc/aur2ccr/aur2ccr.conf.
.sp
The pacman configuration file for Arch repositories is in /etc/aur2ccr/archrepos.pacman.conf
.sp
All configuration files can also be placed in '~/.aur2ccr/'. In fact, settings in these 'local' configuration files take precedence over global configurations. 
.sp

.SH "REPORTING BUGS"
.sp
Please report bugs at: <https://github.com/ccr-tools/aur2ccr/issues/>


.SH "AUTHORS"
.sp
george, Inkane, CaptainKnots


aur2ccr.py
#!/usr/bin/env python2

from __future__ import print_function
from __future__ import unicode_literals

import argparse


def main():
    parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter,
            description="Usage: aur2ccr [OPTIONS] [PACKAGES]\n"
            "Create CCR-ready source packages from packages in AUR or Arch repos",
            epilog="Report bugs at"
            ": <https://github.com/ccr-tools/aur2ccr/issues/>\n"
            "aur2ccr home: <https://github.com/ccr-tools/aur2ccr/>")
    parser.add_argument("--maintainer", "-m", help="add maintainer information "
            "to each PKGBUILD before building")
    parser.add_argument("-e", "--edit",
            help="edit the PKGBUILD for each package with $EDITOR before building ",
            action="store_true")
    parser.add_argument("-c", "--consolidate", help="move all CCR source packages "
            "into the working directory", action="store_true")
    parser.add_argument("-i", "--install", help="install packages that build "
            "successfully, in order to test them", action="store_true")
    parser.add_argument("-f", "--from-src", help="create a CCR source package "
            "from source package <file>", metavar="<file>")
    parser.add_argument("-u", "--from-url", metavar="<url>",
            help="... or from an AUR/CCR-compatible source package at <URL>")
    parser.add_argument("-d", "--from-dir", metavar="<dir>",
            help="...or from a PKGBUILD and (optional) other files in <dir>")
    parser.add_argument("-C", "--cd", metavar="<newdir>",
            help="use <newdir> as the working directory instead of the current dir")
    parser.add_argument("-t", "--temp", help="use $tmpdir as the build directory "
            "instead of the current dir (all finished src packgages will be "
            "moved into the current dir, like -c)", action="store_true")
    parser.add_argument("-l", "--log", metavar="<logfile>",
            help="log all messages from aur2ccr in <logfile>")
    parser.add_argument("-s", "--setup", action="store_true",
            help="perform setup related tasks (not required)")
    parser.add_argument("-a", "--addad", action="store_true",
            help="add an aur2ccr advertisment - as an Contributor")
    parser.add_argument("-v", "--version", action="store_true",
            help="print version information and exit")
    return parser


if __name__ == "__main__":
    parser = main()
    parser.parse_args()
